receiver/cloudflarereceiver/logs.go

```diff
@@ -0,0 +1,292 @@
+	l.logger.Debug("starting receiver HTTP server")
+	// We use l.server.Serve* over l.server.ListenAndServe*
+	// So that we can catch and return errors relating to binding to network interface on start.
+	var lc net.ListenConfig
+
+	listener, err := lc.Listen(ctx, "tcp", l.cfg.Endpoint)
+	if err != nil {
+		return err
+	}
+
+	l.wg.Add(1)
+	go func() {
+		defer l.wg.Done()
+
+		l.logger.Debug("Starting ServeTLS",
+			zap.String("address", l.cfg.Endpoint),
+			zap.String("certfile", l.cfg.TLS.CertFile),
+			zap.String("keyfile", l.cfg.TLS.KeyFile))
+
+		err := l.server.ServeTLS(listener, l.cfg.TLS.CertFile, l.cfg.TLS.KeyFile)
+
+		l.logger.Debug("Serve TLS done")
+
+		if err != http.ErrServerClosed {
+			l.logger.Error("ServeTLS failed", zap.Error(err))
+			host.ReportFatalError(err)
+		}
+	}()
+	return nil
+}
+
+func (l *logsReceiver) handleRequest(rw http.ResponseWriter, req *http.Request) {
+	if len(l.cfg.Secret) > 0 {
+		secretHeader := req.Header.Get(secretHeaderName)
+		if secretHeader == "" {
+			rw.WriteHeader(http.StatusUnauthorized)
+			l.logger.Debug("Got payload with no Secret when it was specified in config, dropping...")
+			return
+		} else if secretHeader != l.cfg.Secret {
+			rw.WriteHeader(http.StatusUnauthorized)
+			l.logger.Debug("Got payload with invalid Secret, dropping...")
+			return
+		}
+	}
+
+	var payload []byte
+	if req.Header.Get("Content-Encoding") == "gzip" {
+		reader, err := gzip.NewReader(req.Body)
+		if err != nil {
+			rw.WriteHeader(http.StatusUnprocessableEntity)
+			l.logger.Debug("Got payload with gzip, but failed to read", zap.Error(err))
+			return
+		}
+		defer reader.Close()
+		// Read the decompressed response body
+		payload, err = io.ReadAll(reader)
+		if err != nil {
+			rw.WriteHeader(http.StatusUnprocessableEntity)
+			l.logger.Debug("Got payload with gzip, but failed to read", zap.Error(err))
+			return
+		}
+	} else {
+		var err error
+		payload, err = io.ReadAll(req.Body)
+		if err != nil {
+			rw.WriteHeader(http.StatusUnprocessableEntity)
+			l.logger.Debug("Failed to read alerts payload", zap.Error(err), zap.String("remote", req.RemoteAddr))
+			return
+		}
+	}
+
+	strPayload := string(payload)
+
+	if strPayload == "test" {
+		l.logger.Info("Received test request from Cloudflare")
+		rw.WriteHeader(http.StatusOK)
+		return
+	}
+
+	logs, err := parsePayload(strPayload)
+	if err != nil {
+		rw.WriteHeader(http.StatusUnprocessableEntity)
+		l.logger.Error("Failed to convert cloudflare request payload to maps", zap.Error(err))
+		return
+	}
+
+	if err := l.consumer.ConsumeLogs(req.Context(), l.processLogs(pcommon.NewTimestampFromTime(time.Now()), logs)); err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		l.logger.Error("Failed to consumer alert as log", zap.Error(err))
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
+}
+
+func parsePayload(payload string) ([]map[string]interface{}, error) {
+	var logs []map[string]interface{}
+	for _, line := range strings.Split(payload, "\n") {
```

Comment:
```suggestion
	// TODO var newline = []byte("\n")
	for _, line := range bytes.Split(payload, newline) { 
```

Reply:
I'm recommending these string/byte changes to avoid unnecessary conversions back and forth.